articles/speakeasy-two-factor-authentication-in-nodejs/index.md

```diff
@@ -0,0 +1,245 @@
+    "start": "nodemon start"
+  },
+  "keywords": [],
+  "author": "mia-roberts",
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.17.1",
+    "node-json-db": "^1.3.0",
+    "speakeasy": "^2.0.0",
+    "uuid": "^8.3.2"
+  }
+}
+```
+
+### Installing the dependencies
+After the setup, our application needs some project-specific dependencies. The dependencies include:
++ `Express` as the backend server.
++ `uuid` to generate universal unique user-id.
++ `node-json-db` as our database to store the data as json.
++ `speakeasy` library for authetication
++ `nodemon` for use as our development dependency. With nodemon, we won't have to restart our application every time we make a change.
+
+```bash
+npm install –save express, nodemon, speakeasy, uuid, node-json-db
+```
+
+### Setting up the application
+First we will set up our server. The scripts below will will help us set up our server. Our server will run on the application's entry point `index.js` file. So add the below block of code in the index file.
+```js
+const express = require('express')
+const app = express();
+app.use(express.json())
+const PORT = process.env.PORT || 5000
+
+```
+After setting the server, we bring our dependencies into the `index.js` file where the server script runs. The `uuid` will create a unique user id, while the `node-json-db` will serve as the database to store the user-id and the secret associated with that user-id.
+```js
+//adding the speakeasy library
+const speakeasy = require('speakeasy')
+
+//adding the uuid dependacy
+const uuid = require('uuid')
+
+//adding node-json-db
+const { JsonDB } = require('node-json-db')
+const { Config } = require('node-json-db/dist/lib/JsonDBConfig')
+```
+
+### The node json databse
+Our application will use the node json database to store user records in json format. Check out  [this link](https://www.npmjs.com/package/node-json-db?S_TACT=AGRAK616). To find more about `node-json-db`. From the documentation of `node-json-db`, to initialize the `node-json-db`, we will add the script below.
+```js
+
+// instance of the node-json-db
+const db = new JsonDB(new Config("DataBase", true, false, '/'));
+```
+ * `new Config`  - Creates a node-json-db database config
+ * `DataBase` - specifies the name of the JSON storage file
+ * `true` Tells the database to save on each push.
+ * `false` Instructs the database to save the database in a human-readable format
+ * `/` - the separator to use when accessing  our database values
+
+
+
+### Sending requests using postman
+Since our application does not have a frontend, we will use postman for sending the requests to the applications' backend. Postman provides an interface for handling requests that would have otherwise been handled by the HTML. Check out [this guide](https://www.blazemeter.com/blog/how-use-postman-manage-and-execute-your-apis) on how to use postman. You can download postman using [this](https://www.postman.com/downloads/) link. 
```

Comment:
```suggestion
Since our application does not have a frontend, we will use postman for sending the requests to the applications' backend.

Postman provides an interface for handling requests that would have otherwise been handled by the HTML.

Check out [this guide](https://www.blazemeter.com/blog/how-use-postman-manage-and-execute-your-apis) on how to use postman. You can download postman using [this](https://www.postman.com/downloads/) link. 
```